### DISCUSSION TITLE
build fault

### DISCUSSION THREAD

**Discussion by kuolung1** [Joshua-Riek/ubuntu-rockchip#842] [General]
in OrangePi 5 plus , use ubuntu 22.04.3 build 24.04 
use 

sudo ./build.sh --board=orangepi-5-plus --suite=noble --flavor=desktop

command to build new iso , and then get this error 

=============================================
[2024-05-29 17:39:24] lb_build
P: Executing auto/build script.
P: Setting up cleanup function
[2024-05-29 17:39:24] lb_bootstrap
P: Setting up cleanup function
[2024-05-29 17:39:24] lb_bootstrap_cache restore
W: skipping , qemu-static binary arm64 was not found
[2024-05-29 17:39:24] lb_bootstrap_copy
[2024-05-29 17:39:25] lb_bootstrap_cdebootstrap
[2024-05-29 17:39:25] lb_bootstrap_debootstrap
W: skipping , qemu-static binary arm64 was not found
[2024-05-29 17:39:25] lb_bootstrap_cache save
W: skipping , qemu-static binary arm64 was not found
P: Begin unmounting filesystems...
P: Saving caches...
./auto/build: 115: cannot create chroot/etc/default/locale: Directory nonexistent
P: Begin unmounting filesystems...
P: Saving caches...
tar: ./*：無法 stat: 沒有此一檔案或目錄
tar: 由於先前錯誤而以失敗狀態離開
mount: rootfs/dev: mount point does not exist.
Error: in ./scripts/config-image.sh on line 84
Error: in ./build.sh on line 229
=================================================

How to fix it

**Comment by hakotani-o** [comment]
Hello.
I not try build on 22.04 , i build on 24.04.

sudo apt install -y build-essential gcc-aarch64-linux-gnu bison \
          qemu-user-static qemu-system-arm qemu-efi u-boot-tools binfmt-support \
          debootstrap flex libssl-dev bc rsync kmod cpio xz-utils fakeroot parted \
          udev dosfstools uuid-runtime git-lfs device-tree-compiler python2 python3 \
          python-is-python3 fdisk bc debhelper python3-pyelftools python3-setuptools \
          python3-distutils python3-pkg-resources swig libfdt-dev libpython3-dev dctrl-tools

**Comment by kuolung1** [comment]
Hello, after clicking the above button to install all the packages, and then re-executing the program, I got the following message:

sudo ./build.sh --board=orangepi-5-plus --suite=noble --flavor=desktop

mount: rootfs/dev: mount point does not exist.
Error: in ./scripts/config-image.sh on line 84
Error: in ./build.sh on line 229

**Comment by hakotani-o** [comment]
Hello.
Please try this.

sudo rm -rf build/ubuntu-live-build
sudo rm build/ubuntu-24.04-preinstalled-desktop-arm64.rootfs.tar.xz
sudo ./build.sh --board=orangepi-5-plus --suite=noble --flavor=desktop

**Comment by kuolung1** [comment]
Tks  It can build finish

**Comment by kuolung1** [comment]
Hello !! 

I can build finish , but it got a "ubuntu-24.04-preinstalled-desktop-arm64.rootfs.tar.xz" , how can i get a .img file for install

**Comment by hakotani-o** [comment]
Hello.

Please check images directory.

**Comment by kuolung1** [comment]
got it Tks

**Comment by Manu7788** [comment]
@hakotani-o 
got the same issue and your fix didnt work for me. its version 22.04 ubuntu. any ideas ?

**Reply by Manu7788** [reply]
i found a solution: use recommended packages install frist:

```
sudo apt-get install -y build-essential gcc-aarch64-linux-gnu bison \
qemu-user-static qemu-system-arm qemu-efi u-boot-tools binfmt-support \
debootstrap flex libssl-dev bc rsync kmod cpio xz-utils fakeroot parted \
udev dosfstools uuid-runtime git-lfs device-tree-compiler python2 python3 \
python-is-python3 fdisk bc debhelper python3-pyelftools python3-setuptools \
python3-distutils python3-pkg-resources swig libfdt-dev libpython3-dev
```

**Reply by hakotani-o** [reply]
Hello,Manu7788 .

Yes, that's right.
